### PR TITLE
Adds slack notifications on failure + junit reporter

### PR DESCRIPTION
**Pull request by vijetmahabaleshwar-okta**
* We have the travis cron job running ITs daily
* If the tests fail, we need to get notifications on the #eng-devex_alerts slack channel
* junit reporter added so that bacon can process the results (nightly jobs)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -17,3 +17,11 @@ env:
...
 after_success:
+
+notifications:
+  slack:
+    on_pull_requests: false
+    on_success: never
+    on_failure: always
+    rooms:
+     - secure: "YoSSxftifz8V8BIGXQSAJTg+mkKs+QVYrnSJ7U4r1M4oP8QUgzBhgv8oBoA35uUA9YqLUi5kYqF7ClRiGg/UAzECEb9svH3Y1ipCvTr/Sz7DEOA960ZgqiEduP8YZARPxJo6aGUR1t8ZF9t1mVK9lFiNchuvZ6+GqKhxpvCkojE3bFZj+0QIw1OuTFdApR7j+ZTazCB9iE6Wx92Jfad0KS47C9jr7+7IF03eSN7VOsLUNQlP659TTeUMbHZ0IcmhalazTtgDiSRuUjYfBTzhKB/gYAVFfccNNx7fQGUHUu29Qm6JKenyZq1BxNNw/CAMvwFePmBukdVaVTdn5YVLe1v3uERnzYd/vbGHdTW8t1Dh13zXtZA4mERRDpSOeyp1MSmJDBg7iKG4UmtpcpuTHQr/1n12BV2wjMLw3VYdQym8t/uYFsnaKcJf9CSR2Rz9B1laLDIxAgzcyjksupKSmMuT1FQiKDAGQCov5UBxvGpSz7MP4fgSe7/de2ypxnLZvrmgWsuriLXgKgXjtGhhr/BKhXkwZc7rmxjJ0ZJodmLIbePG7ylFH/e4B2rBwApMTr518/Zh6uQIXExxmUq9fNb5c9ObReU04DVded6uqy2bnIQMdjZ3oevuXCdwZiIjAyKpLx3WSsYs3pTfLgLkyGhNkwoqO/NTew7bt6T9l40="
```
Comment (jmelberg-okta) [inline .travis.yml]: Just out of curiosity, any reason this is value is surrounded by double quotes?

All other `secure` values i've seen have omitted them.

File: .travis.yml [hunk 2]
```diff
@@ -17,3 +17,11 @@ env:
   matrix:
     secure: r8OGdpfg+XYSakj6YvFyOm5ll7Dm1h1SW17fhUvmSAFin1nvlTWGiORgt0Jhe6OqKHN97WHRZz6ZWjVPbcHlKQD1Py5u4o7IENXf5BOAX7jKESFXv5Ecbg65b4oQJbUuHDtQzA1hbQwSzU012eByqJhEwQ0hnoJYGP9PUhCg+NaGHpjLiMCf8KpjrugUntFc2zox0CD5MFC60VvtflIe3ix3T5Vupt1evyQ11Nt25++Sd/D212WxVxGWOaSbEr8IQ1QEkwOKXV1k6Vq5dHbaJ+OujVPGSm2HiiJGc/KW7wxjTr/LfEg8WngQpocrws3+gYAWmL28IO+oggitBOFi9Fzy/1xEQHnD3mtbx3P7iFWqMqx6QOadRZyGOrlpkACURpnL0gY76tdhcgs8HtvQO2G/Ht7hhmDkaXXxuImFjiSIe6ruYRzpK691CE3Kw4BmYTq+RDZayYq82l/5uamZbwSzxfQ3J+ZQazSCl9nqFVqV7b8NNM9Xlo3CFy8r5KLZechSIwXQzi/c6YpsoJ5unbLX1T36pGeIXULmcPBNgHstt9IjdJGVjvpolG56bW8G3B12gVoTdCigtUgDvjAYRRF052EsBpU/YCAIA2UZockbfdRPprMqVUGgeOC0vH/OfIWFcVwqudofrzYXRJA3ykXBoYaVwwWEC1i02cn0EUY=
 after_success:
```
Comment (jmelberg-okta) [inline .travis.yml]: nit: Not sure if this is being used, so we can probably remove the `after_success` key here.

File: .travis.yml (hunk 1 above)
Comment (vijetmahabaleshwar-okta) [reply .travis.yml]: I followed the travis documentation from here - https://docs.travis-ci.com/user/notifications/#Configuring-Slack-notifications
In the examples, they have the double quotes for some reason.

File: .travis.yml (hunk 2 above)
Comment (vijetmahabaleshwar-okta) [reply .travis.yml]: Removed

File: package.json [hunk 3]
```diff
@@ -17,8 +17,8 @@
     "jest": "jest --coverage test/jest/*.js",
     "predocs": "rm -rf ./jsdocs && mkdir jsdocs/ && ./utils/make-jsdoc-readme.js > ./jsdocs/jsdoc-temp.md",
     "docs": "./node_modules/.bin/jsdoc src/ -c ./docs/config.json -d ./jsdocs/ -P ./package.json -R ./jsdocs/jsdoc-temp.md -r",
-    "test:integration": "nyc --reporter=text --reporter=html mocha test/it/*.js --no-timeouts",
-    "test:unit": "nyc --reporter=text --reporter=html  mocha test/unit/*.js --no-timeouts",
+    "test:integration": "nyc --reporter=text --reporter=html mocha test/it/*.js --no-timeouts --reporter mocha-junit-reporter",
```
Comment (jmelberg-okta) [inline package.json]: IIRC - this outputs a `./test-results.xml` file with test output. If it does, can you make sure this file gets omitted in our `.gitignore`?

File: .gitignore [hunk 4]
```diff
@@ -11,3 +11,4 @@ jsdoc/
 
 .envrc
 .DS_Store
+test-results.xml
```
Comment (vijetmahabaleshwar-okta) [inline .gitignore]: This is the output of junit reporter.

**Pull request merged**